manuscript/16-CompFound.txt

```diff
@@ -739,18 +650,724 @@ long double, double byte character set (DBCS), Boolean κ.λπ.), καθώς κ
+πολλαπλών επεξεργαστών και χρονοπρογραμματισμό πραγματικού χρόνου, ταυτόχρονο
+προγραμματισμό, αδιέξοδα, κρίσιμες περιοχές, υπό όρους κρίσιμες περιοχές και οθόνες. 
+
+Η επικοινωνία μεταξύ διαφορετικών διεργασιών είναι σημαντική στα πολυδιεργασιακά,
+πολυχρηστικά λειτουργικά συστήματα. Ένας μηχανικός λογισμικού πρέπει να έχει βαθιά
+κατανόηση της επικοινωνίας μεταξύ διεργασιών (IPC) και των τύπων IPC, συμπεριλαμβανομένων
+των μηνυμάτων, των σωλήνων, της κοινής μνήμης, των σεμαφόρων, της αρθρωματοποίησης
+και του συγχρονισμού διεργασιών. 
+
+Χρησιμοποιούνται διάφοροι τύποι μηχανισμών κλειδώματος για να εξασφαλιστεί ο σωστός συγχρονισμός
+δεδομένων μεταξύ διεργασιών, συμπεριλαμβανομένων των ημαφορικών, των δυαδικών ημαφορικών,
+των ημαφορικών καταμέτρησης και των μηχανισμών κλειδώματος mutex. Η βαθιά κατανόηση των κοινών
+προκλήσεων των IPCs, των αδιεξόδων, των σεναρίων αδιεξόδων και του χαρακτηρισμού αδιεξόδων,
+της πρόληψης, της αποφυγής, της ανίχνευσης και της αποκατάστασης αδιεξόδων και των
+γραφημάτων προτεραιότητας είναι κρίσιμη και πρέπει να εσωτερικευτεί με τη βοήθεια
+μελετών περίπτωσης. 
+
+Οι μηχανικοί λογισμικού απαιτείται να μελετήσουν, με παραδείγματα, τις ταυτόχρονες
+γλώσσες, τις διεργασίες και τον προγραμματισμό, τις έννοιες των εργασιών και των
+διεργασιών και τους διάφορους τύπους προγραμματισμού: και αλγόριθμους χρονοπρογραμματισμού
+(first come, first served (FCFS), shortest job first (SJF), shortest remaining time first
+(SRTF), priority scheduling, round robin και συνδυασμένα σχήματα).
+
+### 5.2 Διαχείριση Μνήμης
+
+Ένας μηχανικός λογισμικού χρειάζεται πολύ καλή κατανόηση του τρόπου διαχείρισης
+της μνήμης στο σύστημα και των διαφόρων τύπων μνήμης και των σχετικών εννοιών
+- φυσική μνήμη, εικονική μνήμη, δευτερεύουσα μνήμη, ιεραρχία μνήμης, σύνδεση και
+κατανομή μνήμης.
+
+Οι μηχανικοί πρέπει να κατανοούν τον κατακερματισμό της μνήμης (τόσο τον εξωτερικό
+κατακερματισμό, όσο και τον εσωτερικό κατακερματισμό) και τις διάφορες έννοιες
+διαχείρισης μνήμης, συμπεριλαμβανομένων των μονάδων, της σελιδοποίησης, των πινάκων
+σελίδων, της τμηματοποίησης, της σελιδοποιημένης τμηματοποίησης, της διαχείρισης
+εικονικής μνήμης, της σελιδοποίησης κατά απαίτηση, της αντικατάστασης σελίδων,
+της κατάρρευσης thrashing και της ανταλλαγής. 
+
+Η μνήμη κατανέμεται στις διεργασίες με διάφορους τρόπους - για παράδειγμα, μέσω
+συνεχούς κατανομής, μη συνεχούς κατανομής, δυναμικής κατανομής μνήμης σε τμήματα,
+στατικής ανταλλαγής και επικαλύψεων. 
+
+Η κατανόηση των λογικών διευθύνσεων, των κατατμήσεων, της στατικής έναντι της
+δυναμικής κατανομής μνήμης, της διαχείρισης του ελεύθερου χώρου και του
+ανασυγκροτήματος των μπλοκ μνήμης είναι επίσης σημαντική. 
+
+Καθώς η διαθέσιμη φυσική μνήμη είναι πάντα περιορισμένη, σχεδιάζονται και
+εφαρμόζονται διάφορες στρατηγικές αντικατάστασης σελίδων μνήμης. Αυτές οι
+στρατηγικές περιλαμβάνουν τις στρατηγικές First-in-first-out (FIFO),
```

Comment:
Εδω μεταφράζουμε το First-in-first-out και παραθέτουμε δίπλα σε παρένθεση το FIFO, αντίστοιχα για τα υπόλοιπα.